sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -91,6 +92,56 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     return query.getOne();
   }
 
+  async getPendingAssessment(
+    generatedDate?: Date,
+  ): Promise<StudentAssessment[]> {
+    const processingDate = generatedDate ? generatedDate : addDays(-1);
+    return this.repo.find({
+      select: {
+        id: true,
+        application: {
+          applicationNumber: true,
+          student: {
+            sinValidation: { sin: true },
+            user: { lastName: true, firstName: true },
+            birthDate: true,
+          },
+        },
+        offering: {
+          educationProgram: {
+            name: true,
+            description: true,
+            credentialType: true,
+            cipCode: true,
+            nocCode: true,
+            sabcCode: true,
+            institutionProgramCode: true,
+          },
+          yearOfStudy: true,
+          studyStartDate: true,
+          studyEndDate: true,
+          actualTuitionCosts: true,
+          programRelatedCosts: true,
+          mandatoryFees: true,
+          exceptionalExpenses: true,
+          studyBreaks: { totalFundedWeeks: true },
+        },
+        disbursementSchedules: {
+          id: true,
+          disbursementValues: { id: true, valueCode: true, valueAmount: true },
+        },
+      },
+      relations: {
+        disbursementSchedules: { disbursementValues: true },
+        application: { student: { sinValidation: true, user: true } },
+        offering: { institutionLocation: true, educationProgram: true },
+      },
+      where: {
+        assessmentDate: dateEqualTo(processingDate),
```

Comment:
For this type of operation, I would always prefer doing the inclusive/exclusive check like `date >= today and < tomorrow` instead of trying to get the last millisecond of the day to avoid issues with the date/time precision on DB.

Reply:
Any reply on this one? I would say that it is taking midnight of the next day also here, right?
For instance
```
2022-11-14T00:00:00.000Z
2022-11-15T00:00:00.000Z
```

Reply:
Added TODO